app/code/community/Bolt/Boltpay/Model/OrderFixer.php

```diff
@@ -0,0 +1,359 @@
+    }
+
+    /**
+     * @throws \Exception
+     */
+    protected function notifyAndSaveOrder()
+    {
+        $msg = Mage::helper('boltpay')->__(
+            "There is a price mismatch issue when saving the order, forcing the price from $%s to $%s",
+            $this->originalMagentoGrandTotal,
+            $this->getBoltGrandTotal());
+        $this->magentoOrder->setState(
+            Bolt_Boltpay_Model_Payment::transactionStatusToOrderStatus($this->boltTransaction->status),
+            true,
+            $msg
+        )->save();
+    }
+
+    /**
+     * @param $magentoItems
+     *
+     * @return boolean
+     */
+    protected function doesOrderItemsHaveTheSameSku($magentoItems)
+    {
+        $skus = array();
+        foreach ($magentoItems as $magentoItem) {
+            $skus[] = $magentoItem->getSku();
+        }
+
+        $values = array_count_values($skus);
+        foreach ($values as $key => $value) {
+            if ($value > 1) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param \Mage_Sales_Model_Order_Item $magentoItem
+     * @param                              $boltItem
+     */
+    protected function bugsnagTheItemChange(Mage_Sales_Model_Order_Item $magentoItem, $boltItem)
+    {
+        Mage::helper('boltpay/bugsnag')->notifyException(
+            new Exception(
+                Mage::helper('boltpay')->__(
+                    "The order item %s price of order #%s has been updated from %s to %s.",
+                    $magentoItem->getSku(),
+                    $this->magentoOrder->getIncrementId(),
+                    $magentoItem->getPrice(),
+                    $this->getBoltItemPrice($boltItem)
+                )
+            )
+        );
+    }
+
+    /**
+     * @param \Mage_Sales_Model_Order_Item $magentoItem
+     * @param                              $boltItem
+     */
+    protected function updateOrderItemPrice(Mage_Sales_Model_Order_Item $magentoItem, $boltItem)
+    {
+        $boltItemPrice = $this->getBoltItemPrice($boltItem);
+        $magentoItem->setPrice($boltItemPrice);
+        $magentoItem->setBasePrice($boltItemPrice);
+        $magentoItem->setOriginalPrice($boltItemPrice);
+        $magentoItem->setBaseOriginalPrice($boltItemPrice);
+        $magentoItem->setRowTotal($this->getBoltItemTotalPrice($boltItem));
+        $magentoItem->setBaseRowTotal($this->getBoltItemTotalPrice($boltItem));
+    }
+
+    /**
+     * @return boolean
+     */
+    protected function overrideMagnetoOrderOnMismatch()
+    {
+        return Mage::getStoreConfigFlag('payment/boltpay/override_magento_order_on_mismatch');
+    }
+
+    /**
+     * @return float
+     */
+    protected function getMismatchPriceToleranceConfig()
+    {
+        return Mage::getStoreConfig('payment/boltpay/mismatch_price_tolerance');
+    }
+
+    /**
+     * @return float
+     */
+    protected function getMismatchPriceTolerance()
```

Comment:
@johnmccombs1 This all looks excellent if this was renamed to maybe `getMismatchPriceDifference` or `getPriceDifference`